clothcalc_placeholder.js

```diff
@@ -6385,42 +6404,31 @@
 
                     }
                     if (Settings.get("collector", true)) {
-                        GameInject.injectItem("Trader", "collector",
-                            function (e) {
-                                return s(e)
-                            });
-                        GameInject
-                            .injectTrader(
-                                "collector",
-                                function (e) {
-
-                                    if (t.isNewItem(e.item_id)) {
-
-                                        var imgInject = '<img src="' + Images.iconNew + '" class="TWDBcollector" title=" #NOTATINV# " ' + ' style="position:absolute;top:0px;left:0px;padding:0px;border:0px;margin:0px;" />'
-
-                                        return imgInject;
-                                    }
-                                    return ""
-                                });
-
-                        GameInject.injectMarket("collector",
-                            function (e) {
-                                return o(e)
-                            })
+                        GameInject.injectItem("Trader", "collector", function (e) {
+                            return s(e)
+                        });
+                        GameInject.injectTrader("collector", function (e) {
+							if (t.isNewItem(e.item_id)) {
+								var imgInject = '<img src="' + Images.iconNew + '" class="TWDBcollector" title=" #NOTATINV# " ' + ' style="position:absolute;top:0px;left:0px;padding:0px;border:0px;margin:0px;" />'
+								return imgInject;
+						}
+							return "";
+						});
+                        GameInject.injectMarket("collector", function (e) {
+                            return o(e)
+                        });
+                        GameInject.injectGetBids();
                     }
                     r.ready = true
                 };
                 r = Loader.add("Collector", "tw-db Collector", i, {
                     Settings: true
                 });
                 t.isNewItem = function (e) {
-                    var t = 0;
                     var n = w.ItemManager.get(e);
-                    var r = w.Bag.getItemByItemId(n.item_id);
-                    var i = w.Wear.wear[n.type];
-                    if (r || i && i.obj.item_id == n.item_id) {
-                        t = (r !== undefined ? r.count : 0) + (i !== undefined && i.obj.item_id == n.item_id ? 1 : 0)
-                    }
+                    var inv = w.Bag.getItemsIdsByBaseItemId(n.item_base_id);
+                    var equip = w.Wear.wear[n.type];
+                    var t = inv.length + (equip && equip.obj.item_base_id == n.item_base_id ? 1 : 0) + (TWDB.ClothCalc.bids.indexOf(n.item_id) > -1 ? 1 : 0)
```

Comment:
Suggestion: I have no idea why was this simple condition previously coded in such an obscure way... You certainly improved the quality of this section of code, but if you want to go a bit further, you could get rid of `t` and just make it a simple condition.
```
var itemEquipped = equip && equip.obj.item_base_id == n.item_base_id;
var itemHasBid = TWDB.ClothCalc.bids.indexOf(n.item_id) > -1;
if (inv.length || itemEquipped  || itemHasBid) {
    return true;
} else { ...
```